src/main/java/no/ssb/dlp/pseudo/service/pseudo/RecordMapProcessorFactory.java

```diff
@@ -144,10 +145,22 @@ private String  process(PseudoOperation operation,
                             .encryptionAlgorithmParameters(funcDeclaration.getArgs())
                             .build());
                 }
+                return mappedValue;
+
+            } else if (operation == DEPSEUDONYMIZE) {
+                PseudoFuncOutput output = match.getFunc().restore(PseudoFuncInput.of(varValue));
                 output.getWarnings().forEach(metadataProcessor::addLog);
+                final String mappedValue = (String) output.getFirstValue();
+                if (isSidMapping && varValue.equals(mappedValue)) {
```

Comment:
I know this is not done in this PR, but we should consider extracting the SID-specific handling in a separate private method, to make the process method cleaner and maybe easier to read.

Reply:
Good suggestion. Maybe this will turn out better when we refactor the map-sid functionality.